File: sdk-python/examples/basic_usage.py
#!/usr/bin/env python3
"""Basic usage example for Ask-a-Human SDK.

This example demonstrates:
- Creating a client
- Submitting a question
- Polling for responses
- Processing the results

Usage:
    export ASK_A_HUMAN_AGENT_ID="my-agent"
    python basic_usage.py
"""

import time

from ask_a_human import AskHumanClient
from ask_a_human.exceptions import AskHumanError


def main() -> None:
    """Run the basic usage example."""
    # Create a client
    # You can also pass base_url and agent_id directly
    client = AskHumanClient(agent_id="example-agent")

    try:
        # Submit a text question
        print("Submitting question to humans...")
        submission = client.submit_question(
            prompt=(
                "Should error messages in a shopping app apologize to the user, "
                "or just state the facts? Consider the emotional impact on users "
                "during payment failures."
            ),
            type="text",
            audience=["product", "creative"],
            min_responses=3,
            timeout_seconds=3600,  # 1 hour
        )

        print("Question submitted!")
        print(f"  Question ID: {submission.question_id}")
        print(f"  Status: {submission.status}")
        print(f"  Expires at: {submission.expires_at}")
        print(f"  Poll URL: {submission.poll_url}")
        print()

        # Poll for responses
        print("Waiting for human responses...")
        print("(In a real application, you would do other work while waiting)")
        print()

        max_polls = 10
        poll_interval = 30  # seconds

        for i in range(max_polls):
            # Check for responses
            response = client.get_question(submission.question_id)

            print(f"Poll {i + 1}/{max_polls}:")
            print(f"  Status: {response.status}")
            print(f"  Responses: {response.current_responses}/{response.required_responses}")

            # If we have enough responses or question is done, process them
            if response.status in ("CLOSED", "EXPIRED") or response.current_responses >= 3:
                print()
                print("=" * 50)
                print("RESULTS")
                print("=" * 50)
                print()

                if response.responses:
                    for j, r in enumerate(response.responses, 1):
                        print(f"Response {j}:")
                        print(f"  Answer: {r.answer}")
                        print(f"  Confidence: {r.confidence}/5")
                        print()
                else:
                    print("No responses received yet.")

                break

            # Wait before next poll
            print(f"  Waiting {poll_interval}s before next poll...")
            time.sleep(poll_interval)
        else:
            print("Reached maximum polls without getting enough responses.")
            print("You can continue polling later using the question_id.")

    except AskHumanError as e:
        print(f"Error: {e}")

    finally:
        client.close()


if __name__ == "__main__":
    main()

File: sdk-typescript/examples/basic-usage.ts
/**
 * Basic usage example for Ask-a-Human SDK.
 *
 * This example demonstrates submitting a question and polling for responses.
 *
 * Run with: npx ts-node examples/basic-usage.ts
 */

import { AskHumanClient } from "../src/index.js";

async function main() {
  // Create a client
  const client = new AskHumanClient({
    agentId: "example-agent",
    // baseUrl: "https://api.ask-a-human.com", // Optional, uses default
  });

  console.log("Submitting a question to Ask-a-Human...\n");

  try {
    // Submit a text question
    const submission = await client.submitQuestion({
      prompt: "Should error messages in software apologize to users, or just state the facts?",
      type: "text",
      audience: ["product", "creative"],
      minResponses: 5,
      timeoutSeconds: 3600, // 1 hour
    });

    console.log(`Question submitted!`);
    console.log(`  ID: ${submission.questionId}`);
    console.log(`  Status: ${submission.status}`);
    console.log(`  Poll URL: ${submission.pollUrl}`);
    console.log(`  Expires: ${submission.expiresAt}`);
    console.log();

    // Poll for responses (simple approach - in production use the Orchestrator)
    console.log("Checking for responses...\n");

    const response = await client.getQuestion(submission.questionId);

    console.log(`Question status: ${response.status}`);
    console.log(`Responses: ${response.currentResponses}/${response.requiredResponses}`);

    if (response.responses.length > 0) {
      console.log("\nHuman responses:");
      for (const r of response.responses) {
        console.log(`  - "${r.answer}" (confidence: ${r.confidence ?? "N/A"})`);
      }
    } else {
      console.log("\nNo responses yet. Check back later!");
    }
  } catch (error) {
    console.error("Error:", error);
    process.exit(1);
  }
}

main();
